sys/shell/cmds: add MTD command from `mtd_raw` test (#21489)
RIOT-OS/RIOT: sys/shell/cmds/mtd.doc.md

@defgroup   sys_shell_commands_mtd MTD shell command

> @ingroup    sys_shell_commands

Reviewer: ```suggestion
@addtogroup sys_shell_commands
```

`ingroup` has a higher priority than `addtogroup`, therefore odd issues can arise. In this case, this page is simply not added to the `sys_shell_commands` page.

@brief      MTD shell command for low level storage device access

> Available mtd sub commands

Reviewer: ```suggestion
Available MTD Subcommands
```

==========================

> There are different sub commands available to read and write data to and from an MTD.
> Additional subcommands for example print information or control power states.

Reviewer: ```suggestion
There are different subcommands available to read and write data to and from an MTD,
as well as additional subcommands for example to print information or to control
power states.
```

> Subcommand: read <addr> <len>

Reviewer: ```suggestion
Subcommand: read \<addr\> \<len\>
```

The tags have to be escaped, otherwise Doxygen tries to interpret them as HTML 😅 

```
cbuec@W11nMate:~/RIOTstuff/riot-gettingstarted/RIOT$ make doc |& grep "mtd.doc.md"
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:11: warning: Unsupported xml/html tag <addr> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:11: warning: Unsupported xml/html tag <len> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:17: warning: Unsupported xml/html tag <page> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:17: warning: Unsupported xml/html tag <offset> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:17: warning: Unsupported xml/html tag <len> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:24: warning: Unsupported xml/html tag <addr> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:24: warning: Unsupported xml/html tag <data> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:32: warning: Unsupported xml/html tag <page> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:32: warning: Unsupported xml/html tag <offset> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:32: warning: Unsupported xml/html tag <data> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:40: warning: Unsupported xml/html tag <page> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:40: warning: Unsupported xml/html tag <offset> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:40: warning: Unsupported xml/html tag <data> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:49: warning: Unsupported xml/html tag <addr> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:49: warning: Unsupported xml/html tag <len> found
/home/cbuec/RIOTstuff/riot-gettingstarted/RIOT/sys/shell/cmds/mtd.doc.md:55: warning: Unsupported xml/html tag <sector> found
```

-----------------------------

> The subcommand consumes a memory address and a length in bytes of how many bytes to read.

Reviewer: ```suggestion
The subcommand takes a memory address and a length in bytes of how many bytes to read.
```

I think "takes" is a better word than "consumes". Same for the other paragraphs.

@see mtd_read

Subcommand: read_page <page> <offset> <len>
-------------------------------------------

> The subcommand consumes a page number, an offset withing that page and a length in bytes

Reviewer: ```suggestion
The subcommand consumes a page number, an offset within that page and a length in bytes
```

@defgroup   sys_shell_commands_mtd MTD shell command

> @addtogroup sys_shell_commands

Reviewer 2: ```suggestion
@ingroup    sys_shell_commands
```

In particular, wouldn't that fix it?

Reviewer: That was the original approach: https://github.com/RIOT-OS/RIOT/pull/21489#discussion_r2091014174

It had some kind of issue, but looking at the CI generated documentation, it looks fine. Idk if I just didn't see it: https://ci.riot-os.org/results/52dc560e1cde43aa9552926a595909a7/doc-preview/group__sys__shell__commands.html

My bad...
